sunpy 2022 gsoc projects (#299)
OpenAstronomy/openastronomy.github.io: _projects/2022/sunpy/database.md

---
name: Database as a data client
desc: Creating a new database module
requirements:
 - Familiarity with Python
difficulty: high
issues:

> - https://github.com/sunpy/sunpy/issues?q=is%3Aissue+is%3Aopen+database

Author: We will need a proper issue for this.

mentors:
 - cadair

>  - nabobalis

Author: ```suggestion
 - wtbarnes
 - nabobalis
```

initiatives:
 - GSOC
project_size:
 - medium

>  - large

Author: I was unable to decide.

Reviewer: if it would work with either scope I think it's OK to keep both. We were told that if we go for large project only we can expect fewer slots than if it's a mix, etc. (We very much don't know yet how this all will work with all the changes to the student selection this year).

Over the years, this module has seen little development and has not been linked to the new `sunpy` downloader, Fido. This have made it difficult to support.

For an overview of the current database, you can read:
<https://docs.sunpy.org/en/stable/guide/acquiring_data/database.html> and <https://docs.sunpy.org/en/stable/guide/tour.html#database-package>

The project has several aims:
A creation of a new sub-package, `sunpy.net.database`.
When a user does a data search and download, the search and file information will be saved to a local database.
The user can then do the same search and will be informed that the files already exist locally avoiding doing the original download again.
The ability for someone to pass a directory of data files and to create a new database with all the metadata information in those data files.

Overall, the code will new but we hope that parts of the original database module could be reused with some tidy up to the source code.

More technical details can be found on the linked issue.

> #### Milestones

Author: ```suggestion

#### Milestones
```

---
name: Database as a data client
desc: Creating a new database module
requirements:

>  - Familiarity with Python

Reviewer 2: ```suggestion
 - Familiarity with Python
 - Familiarity with SQLAlchemy
```

difficulty: high
issues:
- https://github.com/sunpy/sunpy/issues?q=is%3Aissue+is%3Aopen+database
mentors:
 - cadair
 - wtbarnes
 - nabobalis
initiatives:
 - GSOC
project_size:
 - medium
 - large
tags:
 - python
collaborating_projects:

>  - sunpy

Reviewer 2: ```suggestion
 - sunpy
 - parfive
```

---
name: Database as a data client
desc: Creating a new database module
requirements:
 - Familiarity with Python
 - Familiarity with SQLAlchemy
difficulty: high
issues:
- https://github.com/sunpy/sunpy/issues/5880
mentors:
 - cadair
 - wtbarnes
 - nabobalis
initiatives:
 - GSOC
project_size:
 - large
tags:
 - python
collaborating_projects:
 - sunpy
 - parfive
---

#### Description

> This project aims to design and implement a new database module within `sunpy`.
> Currently the database module was created to manage data files that are stored locally or on a remote server
> you have access/control to.
> 
> You are be able to add/remove files, fetch missing files from various data sources.
> Within the the `sunpy` documentation we have overviews of database: <https://docs.sunpy.org/en/stable/guide/acquiring_data/database.html> and <https://docs.sunpy.org/en/stable/guide/tour.html#database-package>
> 
> However time, the code has been left to "rot" and it has no real integration with the new `sunpy` downloader,
> Fido.
> These problems have made its original use case difficult to support anymore.
> 
> The goal of this project is to create a "database2" module that will take the idea (and most of the code)
> behind the original database that integrates with Fido in a manner that allows you to search your data archive as if
> it was a data provider.
> This will be called a "database client".
> 
> In addition, we will want to replace parts of the original database functionally with more maintainable and testable code.

Author: ```suggestion
This project aims to design and implement a new database sub-package within `sunpy`.
Currently, there is a `sunpy.database` that was created to manage locally data files stored locally or on a remote server with direct access. This allows someone to manage the files, to add missing files, to remove them and to search based on specific metadata stored in the file.
Over the years, this module has seen little development and has not been linked to the new `sunpy` downloader, Fido. This have made it difficult to support.

For an overview of the current database, you can read:
<https://docs.sunpy.org/en/stable/guide/acquiring_data/database.html> and <https://docs.sunpy.org/en/stable/guide/tour.html#database-package>

The project has several aims:
A creation of a new sub-package, `sunpy.net.database`.
When a user does a data search and download, the search and file information will be saved to a local database.
The user can then do the same search and will be informed that the files already exist locally avoiding doing the original download again.
The ability for someone to pass a directory of data files and to create a new database with all the metadata information in those data files.

Overall, the code will new but we hope that parts of the original database module could be reused with some tidy up to the source code.

More technical details can be found on the linked issue.

```

> 
> 

Author: ```suggestion

```

---
name: Database as a data client
desc: Creating a new database module
requirements:
 - Familiarity with Python
 - Familiarity with SQLAlchemy
difficulty: high
issues:

> - https://github.com/sunpy/sunpy/issues?q=is%3Aissue+is%3Aopen+database

Author: ```suggestion
- https://github.com/sunpy/sunpy/issues/5880
```

mentors:
 - cadair
 - wtbarnes
 - nabobalis
initiatives:
 - GSOC
project_size:

>  - medium

Author: ```suggestion
```
